releases.md

```diff
@@ -0,0 +1,195 @@
+organisation on npmjs.com so they are able to publish the release package
+(see Step 6).
+
+## When to publish a release
+
+[Link to another document here?]
+
+## How to publish a release
+
+Before publishing, the following preconditions should be met:
+- A release proposal issue or tracking pull request (see "Proposal branch"
+  below) will exist documenting:
+  - the proposed changes
+  - the type of release: patch, minor or major
+  - the version number (according to semantic versioning - http://semver.org)
+- The proposed changes should be complete.
+[Link to document for more info on proposing a release etc]
+[Add "For more information..." link here to another document (contributor
+ guide?), for more detailed information about the general
+ development/contribution process works? Perhaps move some of this detail there
+ as well?]
+
+There are two main release flows: patch and non-patch.
+
+The patch flow is for making **patch releases** of the **current version of
+Express**. As per semantic versioning, patch releases are for simple changes,
+eg: typo fixes, patch dependency updates, and simple/low-risk bug fixes.
+Every other type of change is made via the non-patch flow.
+
+### Patch flow
+
+In the patch flow, simple changes are committed to the `master` branch which
+acts as an ever-present branch for the next patch release of the current major
+version of Express.
+
+"Main development branch"
+- For the current major version of Express there is a branch in git named
+  `master`.
+- This branch contains the completed commits for the next patch release of the
+  current major version.
+- Patch releases for the current major version are published from this branch.
+
+`master` is usually kept in a state where it is ready to release. Releases are
+made when sufficient time or change has been made to warrant it. This is usually
+proposed and decided using a github issue.
+
+### Non-patch flow
+
+In the non-patch flow, changes are committed to a temporary proposal branch
+created specifically for that release proposal. The branch is based on the
+most recent release of the major version of Express that the release targets.
+
+"Release branch"
+- For any given major version of Express (current, previous or next) there is
+  a branch in git for that release named ``<major-version>.x` (eg: `4.x`).
+- This branch contains the code from the most recently published minor or
+  patch version of the release.
+- Releases are published from these branches.
+  [Not true presently, since releases are published from master for current
+   version. Maybe it should be true though, since 4.x is out of date]
+
+"Proposal branch"
+- A branch in git representing a proposed new release of Express. This can be a
+  patch, minor or major release, named `<major-version>.0` for a major release,
+  `<major-version>.<minor-version>` for a minor release, or `<major-version>.<minor-version>.<patch-version>` for a patch release.
```

Comment:
So typically when a version in in release mode, everything is still done the same, but any reference to `master` would just means `{major}.x` instead. For example, if path-level changes are committed directly to `master`, then would be committed directly to `3.x` for Express 3. When a 3.21 is proposed, that is done on a `3.21` branch, with a pull request into `3.x` (instead of `master`). I hope that helps!